### ISSUE TITLE
EOFError exception is not catched

### ISSUE THREAD

**Issue by Phyks** [ansible/ansible#11413]
Hi,

When doing a `^D` without entering anything in the `SUDO password` prompt, an `EOFError` is thrown by `getpass` but not catched.

I think it could be catched and a pretty error message could be displayed instead.

Traceback is:

```
SUDO password: Traceback (most recent call last):
  File "/usr/bin/ansible-playbook", line 324, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/usr/bin/ansible-playbook", line 138, in main
    become_method=prompt_method)
  File "/usr/lib/python2.7/site-packages/ansible/utils/__init__.py", line 1134, in ask_passwords
    becomepass = getpass.getpass(prompt=become_prompt)
  File "/usr/lib/python2.7/getpass.py", line 71, in unix_getpass
    passwd = _raw_input(prompt, stream, input=input)
  File "/usr/lib/python2.7/getpass.py", line 135, in _raw_input
    raise EOFError
EOFError
```

**Comment by bcoca** [comment]
fixed in devel via 0a2a9557b82bbc65813211194faeb00f43c43b40

**Comment by Phyks** [comment]
Thanks!
